### PR TITLE
UI Auth package

### PR DESCRIPTION
**Pull request by conrad-vanl**
### What does this PR do, or why is it needed?

Moves auth-related code to a separate package

### What design trade-offs/decisions were made?

Many things had to be commented out, such as [analytics/tracking code](https://github.com/ApollosProject/apollos-prototype/issues/589)

### How do I test this PR?

Make sure auth related stuff still works, especially side-effects of auth (such as recently liked and liking in general).

### What automated tests did you write?

None. just moved things around.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by conrad-vanl** [comment]
@vinnyjth as you have time, I'd like your help with this. This branch works and tests pass, although there are blocks of code now commented out.

This upgrades to the newest version of apollo-client, and pushes state code for `apollos-ui-auth` into its own package. EXCEPT for the `authLink`, which I haven't figured out how to move into `-auth` yet (likely, `-auth` will just export a preconfigured `authLink` you have to include, but I'm trying to see if there's a magical way to do it in the `AuthProvider` component).

Anyways - would like your thoughts on how I'm organizing changes in this branch, the apollo-client upgrade, and how we might look to packageize analytics.

File: packages/apolloschurchapp/src/Providers.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+import React from 'react';
 import { Providers } from '@apollosproject/ui-kit';
+import { Provider as AuthProvider } from '@apollosproject/ui-auth';
+import NavigationService from './NavigationService';
 import ClientProvider from './client';
 
-export default nest(ClientProvider, Providers);
+const AppProviders = (props) => (
+  <ClientProvider>
+    <AuthProvider onTriggerAuth={() => NavigationService.navigate('Auth')}>
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/Providers.js]: `onTriggerAuth` doesn't feel like the best name we could give this. 
Two thoughts:
- Given the name is `AuthProvider`, I wouldn't expect `AuthConsumer` to give me a function to navigate to Auth. I honestly don't know what I would expect :) Maybe an Auth class? 
- If we ever want to add more possibilities  for what this Provider can do, we'll be locked down without changing the  return value of the Consumer. 

My initial thought is that the provider `value=` should be something like `{ visitAuth: () => {...} }` or similar. In the future we could extend to `{ visitAuth, visitPasswordReset: () => {...}, ...etc }` if needed. 

This is based on the vision I have of in my head of where you are going with this module, which might not be accurate :D Would love to hear your thoughts.

File: packages/apollos-ui-auth/src/Provider.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      return null;
+    },
+  },
+};
+
+const Provider = ({ children, onTriggerAuth }) => (
+  <AuthContext.Provider value={onTriggerAuth}>
+    <ApolloConsumer>
+      {(client) => {
+        client.addResolvers(resolvers);
```
Comment (vinnyjth) [inline packages/apollos-ui-auth/src/Provider.js]: this is totally sweet.

**Comment by vinnyjth** [comment]
@conrad-vanl Left some thoughts after looking through the code. I'm liking the idea more and more of putting analytics in a provider and importing the consumer in the `auth` module. I think it's fine if we merge this PR leaving the `tracks` commented out, assuming we create issues to fix them :)

File: packages/apolloschurchapp/src/client/index.js [hunk 3]
```diff
@@ -6,21 +6,24 @@ import { ApolloLink } from 'apollo-link';
...
   cache,
   queryDeduplication: true,
   shouldBatch: true,
+  resolvers,
+  typeDefs: schema,
 });
 
-client.onResetStore(clientStateLink.writeDefaults);
+// client.onResetStore(() => cache.writeData(defaults));
+// cache.writeData(defaults);
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply packages/apolloschurchapp/src/client/index.js]: this was throwing errors uncommented

File: packages/apollos-ui-auth/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      ".+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
+    }
+  },
+  "peerDependencies": {
+    "@apollos-project/ui-kit": "*",
+    "react": "*",
+    "react-native": "*"
+  },
+  "dependencies": {
+    "@apollosproject/ui-kit": "^0.8.0-alpha.8",
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply packages/apollos-ui-auth/package.json]: I _think_ it needs to be in both peer deps and dev deps. Fixed!

File: packages/apolloschurchapp/src/client/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply packages/apolloschurchapp/src/client/index.js]: Figured out proper syntax for this...

**Comment by vinnyjth** [comment]
👁

**Comment by vinnyjth** [comment]
I'm not sure off the top of my head what would change to cause this behavior, but I'm not getting logged out automatically as a result of having invalid credentials (changed my login token after logging in)

**Comment by vinnyjth** [comment]
Another issue I found - uploading a new photo is more inconsistent with this PR. 

I tried this both in master and in this branch, running `yarn install` each time. 

**Master**
1. Upload new image B, image B replaces old image A on the settings screen.
2. On the profile screen, image B is shown.
3. If you navigate back to the settings screen, image A is shown again. 🐛
4. If you navigate back to the profile screen, image B is still shown.
5. Image B is shown in Rock web UI

**ui-auth-package**

1. Upload new image B, image A stays on the settings screen 🐛
2. On the profile screen, image A is shown. 🐛
3. If you navigate back to the settings screen, image A is shown. 🐛
4. If you navigate back to the profile screen, image B is still shown. (weird,. right?)
5. Image B is shown in Rock web UI

**Comment by conrad-vanl** [comment]
@vinnyjth I noticed something similar I think with liking and the new likes not consistently showing up in the "Recently Liked" section. Based on the changes made, I can't figure out why. Any clue?

File: packages/apolloschurchapp/package.json [hunk 5]
```diff
@@ -64,16 +64,16 @@
     }
   },
   "dependencies": {
+    "apollo-cache-inmemory": "1.5.1",
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/package.json]: A lot of these deps got updated...wonder if any of them could be causing the issues we're seeing on their own

**Comment by vinnyjth** [comment]
> @vinnyjth I noticed something similar I think with liking and the new likes not consistently showing up in the "Recently Liked" section. Based on the changes made, I can't figure out why. Any clue?

I actually had no issue with likes 😂 I hadn't finished hardcore testing though. Both those issues have  to do with updating the cache after a mutation, so I'm assuming the issue lies somewhere in that functionality? I'll glance at those changelogs for any ideas.

**Comment by vinnyjth** [comment]
@conrad-vanl I think I merged changes in properly - what do you think the path forward is for this PR? Do you want to tackle figuring out how to resolve those pesky cache bugs or do you want me to dig in?

**Comment by vinnyjth** [comment]
MediaPlayer Snapshots are broken right now, and probably won't be fixed until vNext of `react-apollo` is released. https://github.com/apollographql/react-apollo/blob/master/Changelog.md#vnext

@conrad-vanl if you're okay with that, I can update the snapshots to their failing states and approve.

**Comment by conrad-vanl** [comment]
@vinnyjth i'm good with your changes. I don't think I can do the final review on this...?

**Comment by vinnyjth** [comment]
@conrad-vanl I don't feel like I can either, I've been too deep.... @burkeshartsis will you give this a once over please?

**Comment by vinnyjth** [comment]
@burkeshartsis any timeline on getting a chance to review this? (You don't even have to look at the code, maybe just open it in a sim and give it a once over to catch anything our stale eyes missed). My next body of work (along with another open PR) is based on this code.

**Comment by redreceipt** [comment]
🤘 Looks good! But I did find another UI bug. The likes on the home screen don't update until a refresh, we should probably make changes locally too just for a better UX.

Kapture 2019-03-20 at 10 19 30
